host/CMakeLists.txt

```diff
@@ -452,12 +452,13 @@ add_library(oehostapp INTERFACE)
 target_link_libraries(oehostapp INTERFACE oehost)
 
 if (UNIX)
-  target_link_libraries(oehostapp INTERFACE
-      -rdynamic
+  target_link_libraries(oehost INTERFACE
```

Comment:
Thanks for this security oriented change. Always welcome.


This change is very subtle and I am not sure if I follow it.
Previously noexecstack was applied to oehostapp and thus all enclave hosts would inherit it.
With this PR, noexecstack is applied to oehost and thus oehostapp and in turn all enclave hosts would inherit it.

Can you add a description for what is the subtle difference?

Reply:
If you link against oehost (as any application is welcome to do, I found this out because the stack protector test links against oehost) you don't get the flag.

And more important than the security benefit, with the upstream kernel driver, many enclaves with an executable stack will simply fail to load due to the Intel PSW calling mprotect on every page that gets loaded.

Some more details in this thread https://lkml.org/lkml/2020/2/18/1477

Reply:
Thanks for the explanation and the fix.

The bigger issue seems to be that we expose oehost and oehostapp targets to the user while we advocate only the use of oehostapp. This definitely needs fixing. I would suggest creating a "good first issue" for this and schedule it for v0.9.

Keeping in mind that we consider the host to be malicious, the host could intentionally not honor the execstack bits  in the ELF. Are we safe in such a scenario?

Reply:
@CodeMonkeyLeet, can you confirm, are page protections measured as part of MRENCLAVE?

If so, then a malicious host is not a concern.

Reply:
Yes, EADD measures the page protections for any page added to the enclave. 

The use of `noexecstack` is not a protection for the enclave threat model, but one for the general application model, which is still valuable, much like we enabled strong stack protector by default in the host lib link options.